mne/io/eeglab/eeglab.py

```diff
@@ -872,51 +624,3 @@ def _dol_to_lod(dol):
     """Convert a dict of lists to a list of dicts."""
     return [dict((key, dol[key][ii]) for key in dol.keys())
             for ii in range(len(dol[list(dol.keys())[0]]))]
-
-
-@deprecated('read_annotations_eeglab is deprecated from 0.17 and will be'
-            ' removed in 0.18. Please use mne.read_annotations')
-def read_annotations_eeglab(fname, uint16_codec=None):
-    r"""Create Annotations from EEGLAB file.
-
-    This function reads the event attribute from the EEGLAB
-    structure and makes an :class:`mne.Annotations` object.
-
-    Parameters
-    ----------
-    fname : str | object
-        The path to the (EEGLAB) .set file.
-    uint16_codec : str | None
-        If your \*.set file contains non-ascii characters, sometimes reading
-        it may fail and give rise to error message stating that "buffer is
-        too small". ``uint16_codec`` allows to specify what codec (for example:
-        'latin1' or 'utf-8') should be used when reading character arrays and
-        can therefore help you solve this problem.
-
-    Returns
-    -------
-    annotations : instance of Annotations
-        The annotations present in the file.
-    """
-    eeg = _check_load_mat(fname, uint16_codec=uint16_codec)
-    return _read_annotations_eeglab(eeg)
-
-
-def _strip_to_integer(trigger):
-    """Return only the integer part of a string."""
-    return int("".join([x for x in trigger if x.isdigit()]))
-
-
-def _event_id_func(trigger, event_id, event_id_func, dropped):
```

Comment:
this should not be removed

Reply:
Then there was a problem with the deprecation cycle / tests that were added, it should have thrown an error. Can you add the necessary tests / functions for the new annotations-based code paths?

Reply:
On second thought, I don't think we want the `event_id_func` business anymore